### ISSUE TITLE
inference time for this extremely high?

### ISSUE THREAD

**Issue by jtoy** [ryankiros/neural-storyteller#11]
I tested this on a random image with this code:
        z = generate.load_all()
        s = generate.story(z, args.input)
        print(s)

Im running with 2 k80s NVIDIA cards and a Xenon with 32 cores, it takes 70 minutes to process one image. It seems the majority of the time is spent in load_all.  Is this normal? what kind of inference performance are others getting?

**Comment by YknZhu** [comment]
load_all() takes pretty long but it only need to run once. 70 minutes seem to be a bit too long considering you are using GPU. Did you set your .theanorc to run on GPU?

**Comment by jtoy** [comment]
yeah, I wasnt in gpu mode. I fixed that, but now when I run the same exact code I get: 

```
s = generate.story(z, args.input)
```

  File "/home/ubuntu/experiment/generate.py", line 42, in story
    feats = compute_features(z['net'], im).flatten()
  File "/home/ubuntu/experiment/generate.py", line 183, in compute_features
    deterministic=True).eval())
  File "/usr/local/lib/python2.7/dist-packages/lasagne/layers/helper.py", line 191, in get_output
    all_outputs[layer] = layer.get_output_for(layer_inputs, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/lasagne/layers/conv.py", line 257, in get_output_for
    conved = self.convolve(input, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/lasagne/layers/corrmm.py", line 144, in convolve
    contiguous_filters = gpu_contiguous(filters)
  File "/usr/local/lib/python2.7/dist-packages/theano/gof/op.py", line 611, in **call**
    node = self.make_node(_inputs, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/theano/sandbox/cuda/basic_ops.py", line 3963, in make_node
    input = as_cuda_ndarray_variable(input)
  File "/usr/local/lib/python2.7/dist-packages/theano/sandbox/cuda/basic_ops.py", line 47, in as_cuda_ndarray_variable
    return gpu_from_host(tensor_x)
  File "/usr/local/lib/python2.7/dist-packages/theano/gof/op.py", line 611, in **call**
    node = self.make_node(_inputs, *_kwargs)
  File "/usr/local/lib/python2.7/dist-packages/theano/sandbox/cuda/basic_ops.py", line 140, in make_node
    dtype=x.dtype)()])
  File "/usr/local/lib/python2.7/dist-packages/theano/sandbox/cuda/type.py", line 93, in **init**
    (self.**class**.**name**, dtype, name))
TypeError: CudaNdarrayType only supports dtype float32 for now. Tried using dtype float64 for variable None

I dont suppose that is normal? this same code works fine in gpu mode. do i need to convert any of the files or do anything else to make it work in gpu mode?

**Comment by jtoy** [comment]
Any way I can see the .theanorc  file you use? I just force gpu mode with:
THEANO_FLAGS=device=gpu
not sure if I need to tweak other settings.

**Comment by jtoy** [comment]
got it all working, needed floatX=float32  ; it went from 70 mins to 6 mins, not bad :)
